csr.py

```
@@ -241,76 +241,283 @@
         valid = {}
         fields = ('C', 'ST', 'L', 'O', 'OU', 'CN', 'keySize', 'keyType', 'curve', 'subjectAltNames', 'allowPrivateDomains')
         required = ('CN',)
-        
+
+import unicodedata # For Unicode normalization
+
+class CsrGenerator(object):
```

Ruff (0.11.9)

Redefinition of unused `CsrGenerator` from line 34

(F811)